### PR TITLE
MTD ETL TDR Geometry update of structure

### PR DESCRIPTION
**Pull request by icosivi**
#### PR description:

A DD4hep version of the `DDMTDLinear` plugin is implemented. Such plugin is used to algorithmically place volumes on ETL `DiscSectors`.
The logical parts `LGAD_2`, `LGAD_r` and `LGAD_r_2` are removed from etl.xml: `LGAD_2` is identical to the `LGAD` logical part therefore it is not needed; whereas `LGAD_r` and `LGAD_r_2` are the LGAD volumes on the back face of the ETL disks and they can be replaced by an `LGAD` rotated by 180 degrees around the y-axis.
The name of the LGAD active volume is changed from `LGAD_ActiveThickness` to `LGAD_TimingActive` to be compatible with BTL definition.
The names of the `SensorModule` logical parts are changed to better define their position within an ETL `DiscSector`:
`SensorModule` --> `SensorModule_Front_Right`
`SensorModule_2` --> `SensorModule_Front_Left`
`SensorModule_r` --> `SensorModule_Back_Left`
`SensorModule_r_2` --> `SensorModule_Back_Right`

ETL-TDR_Front-Face

#### PR validation:

`testMTDinDDD.py` and `testMTDinDD4hep.py` allow comparing the DDD and DD4hep geometries. The dump geometries are identical except from the signs of the rotation matrix components (-0.000000 in DDD, 0.000000 in DD4hep).

The new geometry with rotated LGAD volumes on the back faces of the ETL disks is equal to the previous one : this was tested with `testMTDinDDD.py`, veryfiing that volumes' center are unchanged.

The volume-DataID association has changed because we changed the `SensorModule` names. The ideal numbering scheme has been adapted to that.

### CODE REVIEW

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
please test

changes should not be related to this PR

File: Geometry/MTDSimData/data/v2/mtdProdCuts.xml [hunk 1]
```diff
@@ -3,23 +3,23 @@
...
-      <PartSelector path="//LGAD_ActiveThickness"/>
-      <Parameter name="CMSCutsRegion" value="FastTimerRegion" eval="false"/>   
-      <Parameter name="ProdCutsForElectrons" value="0.1*mm"/>                  
-      <Parameter name="ProdCutsForPositrons" value="0.1*mm"/>                  
-      <Parameter name="ProdCutsForGamma" value="0.1*mm"/>                      
+      <PartSelector path="//LGAD_TimingActive"/>
+      <Parameter name="CMSCutsRegion" value="FastTimerRegion" eval="false"/>
+      <Parameter name="ProdCutsForElectrons" value="0.1*mm"/>
+      <Parameter name="ProdCutsForPositrons" value="0.1*mm"/>
+      <Parameter name="ProdCutsForGamma" value="0.1*mm"/>
```
Comment (ianna) [inline Geometry/MTDSimData/data/v2/mtdProdCuts.xml]: what about adding a production cut on Proton equal to the one on Electron: see issue https://github.com/cms-sw/cmssw/issues/27343?

**Review by ianna — CHANGES_REQUESTED** [review]
@fabiocos - wouldn't it be better to create new versions on the xml files?

**Comment by kpedro88** [comment]
@ianna these XML files are only used in the development geometry I12, which is not part of a central production yet. I would prefer not to spawn repeated versions of development geometries. We should allow developers to build up new geometries before enforcing strict versioning.

**Comment by kpedro88** [comment]
+upgrade

File: Geometry/MTDSimData/data/v2/mtdProdCuts.xml (hunk 1 above)
Comment (fabiocos) [reply Geometry/MTDSimData/data/v2/mtdProdCuts.xml]: @ianna thanks for reminding me about this issue, I will update the production cuts in a next round, where I also update the name of some volumes to make the RECO geometry management easier

**Comment by fabiocos** [comment]
@kpedro88 I agree in general with your comment, I believe anyway that scenario D53 is not frozen and that we may further update it as it is strictly speaking "work in progress". In this sense, I would also like to have from @civanch a confirmation that the above production cuts are still meaningful for a silicon sensor that now is 50 microns thick, as the LGAD active area is. I see that tracker for phaseII is moving to a 0.01 mm cut, allowing for secondary production even within very thin layers. I wonder whether we should do the same here for the new ETL (while for BTL, LYSO crystals, the present value looks reasonable).

**Comment by silviodonato** [comment]
please test workflow 23234.1001, 23434.1001

**Comment by fabiocos** [comment]
The observed difference in DQM summary cannot be due to this PR

**Comment by silviodonato** [comment]
+1

**Pull request merged**